src/engine/ruleset/decoders/sonicwall/firewall-general.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: decoder/sonicwall-firewall-general/0
+
+metadata:
+  module: firewall
+  title: Firewall logs decoder
+  description: Decoder for Linux firewall logs
+  author:
+    name: Wazuh Inc. info@wazuh.com
+    date: 2023-01-12
+  references:
+    - https://www.sonicwall.com/es-mx/support/technical-documentation
+
+sources:
+  - decoder/syslog/0
+  - decoder/queue-localfile/0
+
+check:
+  - event.original: +r_match/id\=.*?sn\=.*?time\=.*?fw
+
+parse:
+  logpar:
+    # Jan  3 13:45:36 192.168.5.1 id=firewall sn=000SERIAL time="2007-01-03 14:48:06" fw=89.160.20.156 pri=6 c=262144 m=98 msg="Connection Opened" n=23419 src=2.2.2.2:36701:WAN dst=89.160.20.156:50000:WAN proto=tcp/50000
+    - event.original: >-
+        <~log.timestamp/date/%b %d %T> <host.ip> <~log.message>
+
+normalize:
+  - check:
+      - ~log.message: +ef_exists
+    logpar:
+      - ~log.message: <~tmp/kv/=/ /"/'>
+  - map:
+      - event.code: $~tmp.m
+      - event.dataset: sonicwall.firewall
+      - event.kind: event
+      - event.module: sonicwall
+      - fileset.name: firewall
+      - input.type: log
+      - observer.product: Firewalls
+      - observer.type: Firewall
+      - observer.vendor: Sonicwall
+      - rsa.internal.messageid: $~tmp.m
+      - rsa.time.event_time: $event.start
+      - service.type: sonicwall
+      - tags: +a_append/forwarded/sonicwall.firewall
+      - \@timestamp: $event.start
+  - check:
+      - ~tmp.src: +ef_exists
+    logpar:
+      - ~tmp.src: <~tmp.src.ip>:<~tmp.src.port>:<~tmp.src.interface.name>
```

Comment:
Please, don't use `logpar` on the `normalize` stage. On the other hand, what if the IP is an IPv6 and it has `:` on it?